src/main/java/edu/unc/lib/boxc/migration/cdm/services/GroupMappingService.java

```diff
@@ -177,17 +143,23 @@ private void generateMultipleGroupMapping(GroupMappingOptions options, Statement
                     matchedValues.add(matchedValue);
                 }
             }
+            // Join matched values when grouping by multiple fields
+            String multipleMatchedValues = null;
+            if (numberGroups > 1 && !matchedValues.isEmpty()) {
```

Comment:
it's fine to join even when there is only field being grouped by, you'll just get back the one value. Even joining on an empty list is often fine, you'll just get an empty string. If we always set a multipleMatchedValues value then you could use that for all the conditions in the `if` down at line 153 instead of using matchedValues some of the time, as well as simplify the number of cases.